### PR TITLE
Skip loading a FHIRPath object if we are older than STU3

### PR DESCRIPTION
**Pull request by tadgh**

### CODE REVIEW

File: hapi-fhir-server-mdm/src/main/java/ca/uhn/fhir/mdm/rules/config/MdmRuleValidator.java [hunk 1]
```diff
@@ -61,7 +62,12 @@
 	public MdmRuleValidator(FhirContext theFhirContext, ISearchParamRetriever theSearchParamRetriever) {
 		myFhirContext = theFhirContext;
 		myTerser = myFhirContext.newTerser();
-		myFhirPath = myFhirContext.newFhirPath();
+		if (myFhirContext.getVersion().getVersion().isEqualOrNewerThan(FhirVersionEnum.DSTU3)) {
+			myFhirPath = myFhirContext.newFhirPath();
+		} else {
+			ourLog.warn("Skipping FHIRPath validation as DSTU2 does not support FHIR");
```
Comment (jamesagnew) [inline hapi-fhir-server-mdm/src/main/java/ca/uhn/fhir/mdm/rules/config/MdmRuleValidator.java:68]: I'm curious about 2 things:

* Is this just papering over a deeper issue? Like, why do we validate FHIRPath here unless we're gonna use it again later? Are we just delaying an inevitable fail?
* Should this be a warning, given that it seems like a routine thing to happen?

Comment (tadgh) [reply hapi-fhir-server-mdm/src/main/java/ca/uhn/fhir/mdm/rules/config/MdmRuleValidator.java:68]: The issue is that in CDR, the BaseAppCtxPersistence is what loads mdmSubmitterConfig. So in DSTU2 mode we load this bean, even though we never actually need it. The real solution is to add a BaseAppCtxPersistenceDSTU3Plus config, which correctly loads the Mdm Submitter beans

Comment (jamesagnew) [reply hapi-fhir-server-mdm/src/main/java/ca/uhn/fhir/mdm/rules/config/MdmRuleValidator.java]: Ah ok - Should the warn be an info then?

Comment (jamesagnew) [reply hapi-fhir-server-mdm/src/main/java/ca/uhn/fhir/mdm/rules/config/MdmRuleValidator.java]: Sorry, I see you did that nevermind

**Review by jamesagnew — APPROVED** [review]

**Pull request merged**